app/src/main/java/app/grapheneos/camera/ui/activities/VideoPlayer.kt

```diff
@@ -35,7 +36,7 @@ class VideoPlayer : AppCompatActivity() {
 
         val videoView = binding.videoPlayer
 
-        val mediaController = object : MediaController(this) {
+        mediaController = object : MediaController(this) {
```

Comment:
Commit message says:
> Solve activity leaked window with mediaController

How does this change fix it?

Reply:
Because below this change, I am overriding onPause to actually hide `mediaController`.

Reply:
```
2022-05-29 03:00:35.506 9930-9930/app.grapheneos.camera.dev E/WindowManager: android.view.WindowLeaked: Activity app.grapheneos.camera.ui.activities.VideoPlayer has leaked window DecorView@1f911ca[] that was originally added here
        at android.view.ViewRootImpl.<init>(ViewRootImpl.java:812)
        at android.view.ViewRootImpl.<init>(ViewRootImpl.java:796)
        at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:390)
        at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:138)
        at android.widget.MediaController.show(MediaController.java:384)
        at android.widget.MediaController.show(MediaController.java:334)
        at app.grapheneos.camera.ui.activities.VideoPlayer$onCreate$mediaController$1.show(VideoPlayer.kt:40)
        at android.widget.VideoView.toggleMediaControlsVisiblity(VideoView.java:790)
        at android.widget.VideoView.onTouchEvent(VideoView.java:730)
        at android.view.View.dispatchTouchEvent(View.java:14599)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3120)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2744)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3120)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2744)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3120)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2744)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3120)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2744)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3120)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2744)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3120)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2744)
        at com.android.internal.policy.DecorView.superDispatchTouchEvent(DecorView.java:502)
        at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1890)
        at android.app.Activity.dispatchTouchEvent(Activity.java:4199)
        at androidx.appcompat.view.WindowCallbackWrapper.dispatchTouchEvent(WindowCallbackWrapper.java:69)
        at com.android.internal.policy.DecorView.dispatchTouchEvent(DecorView.java:460)
        at android.view.View.dispatchPointerEvent(View.java:14858)
        at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:6446)
        at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:6247)
        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:5725)
        at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:5782)
        at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:5748)
        at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:5913)
        at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:5756)
        at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:5970)
        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:5729)
        at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:5782)
        at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:5748)
        at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:5756)
        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:5729)
        at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:8696)
        at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:8647)
        at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:8616)
        at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:8819)
        at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:259)
        at android.os.MessageQueue.nativePollOnce(Native Method)
        at android.os.MessageQueue.next(MessageQueue.java:335)
        at android.os.Looper.loopOnce(Looper.java:161)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7842)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
2022-05-29 03:00:35.506 9930-9930/app.grapheneos.camera.dev E/WindowManager:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
```

Reply:
I see. `MediaController` shows controls in a separate window, and `mediaController.hide()` removes that window.